model/result/ResultService.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA
+ */
+namespace oat\taoLtiConsumer\model\result;
+
+use common_exception_Error;
+use oat\oatbox\event\EventManager;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoLtiConsumer\model\result\parser\XmlResultParser;
+use oat\taoResultServer\models\Exceptions\DuplicateVariableException;
+use Throwable;
+
+class ResultService extends ConfigurableService
+{
+    public const SERVICE_ID = 'taoLtiConsumer/resultService';
+    public const LIS_SCORE_RECEIVE_EVENT = 'LisScoreReceivedEvent';
+    public const DELIVERY_EXECUTION_ID = 'DeliveryExecutionID';
+
+    /**
+     * @param string $payload
+     *
+     * @return mixed
+     * @throws ResultException
+     * @throws Throwable
+     */
+    public function process($payload)
+    {
+        try {
+            $request = $this->getXmlResultParser($payload);
+            $action = $request->getRequestType();
+
+            if (!method_exists($this, $action)) {
```

Comment:
I would advise having explicit `if` or `switch` (to check $action string) and corresponding explicit method call:
1. dynamic call (without checking $action is in allowed range) isn't recommended due to safety reasons: arbitrary method could be executed.
2. dynamic call doesn't allow IDE to index usages of calling method, you can't use "Find Usages" or "Refactor: .." features in that case